presto-iceberg/src/test/java/io/prestosql/plugin/iceberg/TestIcebergSmoke.java

```diff
@@ -593,6 +593,210 @@ private void testPredicating(Session session, FileFormat fileFormat)
     }
 
+    @Test
+    public void testDateTransforms()
+    {
+        testWithAllFileFormats(this::testDateTransformsForFormat);
+    }
+
+    private void testDateTransformsForFormat(Session session, FileFormat format)
+    {
+        testHourTransformForFormat(session, format);
+        testDayTransformForFormat(session, format);
+        testMonthTransformForFormat(session, format);
+        testYearTransformForFormat(session, format);
+    }
+
+    private void testHourTransformForFormat(Session session, FileFormat format)
+    {
+        String select = "SELECT d_hour, row_count, d.min AS d_min, d.max AS d_max, b.min AS b_min, b.max AS b_max FROM \"test_hour_transform$partitions\"";
+
+        assertUpdate(session, format("CREATE TABLE test_hour_transform (d TIMESTAMP, b BIGINT)" +
+                " WITH (format = '%s', partitioning = ARRAY['hour(d)'])", format.name()));
+        assertUpdate(session, "INSERT INTO test_hour_transform VALUES" +
+                        "(TIMESTAMP '2015-01-01 10:01:23', 1)," +
+                        "(TIMESTAMP '2015-01-01 10:10:02', 2)," +
+                        "(TIMESTAMP '2015-01-01 10:55:00', 3)," +
+                        "(TIMESTAMP '2015-05-15 12:05:01', 4)," +
+                        "(TIMESTAMP '2015-05-15 12:21:02', 5)," +
+                        "(TIMESTAMP '2015-02-21 13:11:11', 6)," +
+                        "(TIMESTAMP '2015-02-21 13:12:12', 7)",
+                7);
+
+        assertQuery(session, "SELECT COUNT(*) FROM \"test_hour_transform$partitions\"", "SELECT 3");
+
+        assertQuery(session, "SELECT COUNT(*) FROM test_hour_transform WHERE hour(d) = 10", "SELECT 3");
+        // 394460 = (2015 - 1970) * 365 * 24 + leap day hours + 10
+        // Parquet has min/max for timestamps but Orc does not.
+        String expected = format == FileFormat.PARQUET ? "VALUES(394460, 3, TIMESTAMP '2015-01-01 10:01:23', TIMESTAMP '2015-01-01 10:55:00', 1, 3)" : "VALUES(394460, 3, NULL, NULL, 1, 3)";
+        assertQuery(session, select + " WHERE d_hour = 394460", expected);
+
+        assertQuery(session, "SELECT COUNT(*) FROM test_hour_transform WHERE hour(d) = 12", "SELECT 2");
+        expected = format == FileFormat.PARQUET ? "VALUES(397679, 2, TIMESTAMP '2015-05-15 12:05:01', TIMESTAMP '2015-05-15 12:21:02', 4, 5)" : "VALUES(397679, 2, NULL, NULL, 4, 5)";
+        assertQuery(session, select + " WHERE d_hour = 397679", expected);
+
+        assertQuery(session, "SELECT COUNT(*) FROM test_hour_transform WHERE hour(d) = 13", "SELECT 2");
+        expected = format == FileFormat.PARQUET ? "VALUES(395687, 2, TIMESTAMP '2015-02-21 13:11:11', TIMESTAMP '2015-02-21 13:12:12', 6, 7)" : "VALUES(395687, 2, NULL, NULL, 6, 7)";
+        assertQuery(session, select + " WHERE d_hour = 395687", expected);
+
+        dropTable(session, "test_hour_transform");
+    }
+
+    private void testDayTransformForFormat(Session session, FileFormat format)
```

Comment:
We should test day/month/year transform for timestamp as well. It's fine to do this as a follow up.

Reply:
Will do as a follow-up.